common/src/com/sony/transmitpower/observer/TelephonyStateObserver.java

```diff
@@ -108,8 +106,15 @@ public static boolean isDataActive(@NonNull final Context context)
 
     // for each SIM:
     private final class PhoneStateListenerImpl extends PhoneStateListener {
-        PhoneStateListenerImpl(int subId) {
-            super(subId);
+        // Keep track of the used manager for deregistration purposes.
+        private final TelephonyManager mTelephonyManager;
+
+        PhoneStateListenerImpl(TelephonyManager manager) {
+            mTelephonyManager = manager;
+        }
+
+        TelephonyManager getManager() {
```

Comment:
Instead of exposing the TelephonyManager write a listen method, e.g. something like:
`public void listen(int events) {
     mTelephonyManager.listen(this, events);
 }`

and then call that at listener.getManager().listen (...) below.

Reply:
Yep, that's what I intended to do but didn't want to place `mTelephonyManager` there in the first place.

Reply:
It's fine for this purpose.